packages/stylelint-config/rules/extras.js

```diff
@@ -0,0 +1,87 @@
+module.exports = {
+  rules: {
+    // Reason: Avoid compound selectors creation using nested ampersand SASS syntax.
+    // Main reasons why use this pattern:
+    // - Unification with design system components code style
+    // - Look up easily for the selector using fulltext search in IDEs
+    // - Better orientation in longer component's code when scrolled down
+    // - Makes `no-descending-specificity` stylelint rule work correctly
+    // - IDE's support to click through from selector usage to its source
+    // Docs: https://stylelint.io/user-guide/rules/list/selector-nested-pattern/
+    'selector-nested-pattern': [
+      '(^&:)|(^&\\.)|(^&\\[)',
+      {
+        message: 'Only pseudo class selector, attribute selector and combination with state class is allowed',
+      },
+    ],
+
+    // Reason: Except for utility classes and third-party overrides, !important can be avoided.
+    // Docs: https://stylelint.io/user-guide/rules/list/declaration-no-important
+    'declaration-no-important': true,
+
+    // Reason: Selector maximum class count should be just one but especially form parts/components
+    // need more complex selectors to add styles when validation is applied.
+    // Setting rule to "1" would be uneasy to pass in almost all projects so "2" is clearly just fine.
+    // In case of inevitable more complex selector it is allowed and recommended to locally disable that rule.
+    // Docs: https://stylelint.io/user-guide/rules/list/selector-max-class
+    'selector-max-class': 2,
+
+    // No default value
+    // Reason: IDs are handy for Javascript, not for CSS anymore.
+    // Docs: https://stylelint.io/user-guide/rules/list/selector-max-id
+    'selector-max-id': 0,
+
```

Comment:
So no more `selector-*` rules?

- `selector-max-compound-selectors`
- `selector-max-specificity`
- `selector-max-universal`
- `selector-no-qualifying-type`

Sorry if we agreed on it and I just don't remember.

Reply:
Yes, we agreed on this. That's why provided reason says that it's handy for js, not for css anymore.
We also tested its implementation on other projects and found only one where this rule needs to be turned off but it is an edge-edge case.

Reply:
No, @adamkudrna, you remember well. We kind of missed it so I'm adding these rules at last.